File: c-cpp/draw_triangle.c
#include <stdbool.h>
#include <stdio.h>

int get_size(void);
void print_spaces(int, int);
void print_blocks(int, bool);

int main(void)
{
    int size = get_size();
    for(int rows = 0; rows < size; rows++)
    {
        print_spaces(rows, size);
        printf("#");
        print_blocks(rows, true);
    }

    return 0;
}

int get_size(void)
{
    int size = 0;
    do 
    {
        printf("Size (3-9): ");
        scanf("%i", &size);
    }
    while(size < 1 || size > 9);
    return size;
}

void print_spaces(int rows, int size)
{
    for (int spaces = size - rows; spaces > 0; spaces--)
    {
        printf(" ");
    }
    print_blocks(rows, false);
}

void print_blocks(int rows, bool jump)
{
    for (int blocks = 0; blocks < rows; blocks++)
    {
        printf("#");
    }

    if (jump == true) printf("\n");
}

File: python/draw_triangle.py
def main() -> None:
    size: int = get_size()

    for row in range(size):
        print_spaces(row, size)
        print('#', end='')
        print_blocks(row, True)


def get_size() -> int:
    size: int = 0

    while size < 1 or size > 9:
        try:
            size = int(input('Size (3-9): '))
        except ValueError:
            print('Not an integer')

    return size


def print_spaces(row: int, size: int) -> None:
    spaces: int = row - size + 1

    while spaces < 0:
        print(' ', end='')
        spaces += 1

    print_blocks(row, False)


def print_blocks(row: int, jump: bool) -> None:
    for _ in range(row):
        print('#', end='')

    if jump:
        print()


if __name__ == '__main__':
    main()
